sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -3,209 +3,334 @@
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # -------------------------------------------------------------------------
-
+import os
 import json
-from azure.appconfiguration import AzureAppConfigurationClient
+from typing import Any, Dict, Iterable, Mapping, Optional, overload, List, Tuple, TYPE_CHECKING
+from azure.appconfiguration import (
+    AzureAppConfigurationClient,
+    FeatureFlagConfigurationSetting,
+    SecretReferenceConfigurationSetting
+)
 from azure.keyvault.secrets import SecretClient, KeyVaultSecretIdentifier
-from azure.core.exceptions import ResourceNotFoundError
-from ._settingselector import SettingSelector
-from ._azure_appconfiguration_provider_error import KeyVaultReferenceError
-from ._constants import KEY_VAULT_REFERENCE_CONTENT_TYPE
+from ._models import AzureAppConfigurationKeyVaultOptions, SettingSelector
+from ._constants import (
+    FEATURE_MANAGEMENT_KEY,
+    ServiceFabricEnvironmentVariable,
+    AzureFunctionEnvironmentVariable,
+    AzureWebAppEnvironmentVariable,
+    ContainerAppEnvironmentVariable,
+    KubernetesEnvironmentVariable
+)
 
 from ._user_agent import USER_AGENT
 
+if TYPE_CHECKING:
+    from azure.core.credentials import TokenCredential
+
 
-class AzureAppConfigurationProvider:
+@overload
+def load_provider(
+        endpoint: str,
+        credential: "TokenCredential",
```

Comment:
(We can defer this question until arch board review). Maybe we should make all these arguments keyword only?

Reply:
Took offline and we can discuss it in arch board review.